engine/src/main/java/org/terasology/config/flexible/FlexibleConfigManager.java

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright 2017 MovingBlocks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.terasology.config.flexible;
+
+import org.terasology.engine.SimpleUri;
+
+public interface FlexibleConfigManager {
+    void addFlexibleConfig(SimpleUri flexibleConfigUri, FlexibleConfig fc);
```

Comment:
- use flexibleConfig for the second parameter, similar to the first one.
- the interface could define the behavior when two configs with the same uri are added (throw RTE)

Reply:
When you say "the config can define the behaviour", you mean I should add a "throws RuntimeException" here?

Reply:
Okay, so based on a quick tour of the rest of the FC code, the `throws` clause is added in the javadocs everywhere, and nowhere in the declaration.

Adding it in the code here makes things inconsistent - although we can can obviously go back and add it to the code everywhere.
Thoughts?

Reply:
Sorry for the delay about this.  

I think in this case it is warranted for the interface to show the exception thrown as it is part of the intended behavior of this method AND it diverges from the behavior of a typical map that would just overwrite the previously existing value for a given key.

Reply:
I had not considered the second part - the divergence. Definitely makes way more sense to me now :D